### PR TITLE
ENH: Multiple changes and enhancements

### PR DESCRIPTION
**Pull request by ElDeveloper**
- Adds a permutations parameter.
- Add handling for zero permutations.
- Avoid returning p-value=0 to mimic the behaviour of other
  permutation-based tests.
- Rename trials to permutations.

### CODE REVIEW

File: q2_diversity/plugin_setup.py [hunk 1]
```diff
@@ -251,7 +251,10 @@
 plugin.methods.register_function(
     function=q2_diversity.procrustes_analysis,
     inputs={'reference': PCoAResults, 'other': PCoAResults},
-    parameters={'dimensions': Int % Range(1, None)},
+    parameters={
+        'dimensions': Int % Range(1, None),
+        'permutations': Int % Range(0, None)
```
Comment (thermokarst) [inline q2_diversity/plugin_setup.py:256]:
```suggestion
        'permutations': Int % Range(1, None) | Str % Choices('disable')
```

File: q2_diversity/_procrustes.py [hunk 2]
```diff
@@ -100,11 +98,16 @@ def _procrustes_monte_carlo(reference: np.ndarray,
         if m2 < true_m2:
             trials_below_m2 += 1
 
-    p_val = trials_below_m2 / trials
+    if permutations == 0:
```
Comment (thermokarst) [inline q2_diversity/_procrustes.py:101]:
```suggestion
    if permutations == 'disable':
```

File: q2_diversity/plugin_setup.py [hunk 3]
```diff
@@ -260,9 +263,18 @@
...
         'reference': ('The ordination matrix to which data is fitted to.'),
         'other': ("The ordination matrix that's fitted to the reference "
-                  "ordination.")
+                  "ordination."),
+    },
+    parameter_descriptions={
+        'dimensions': ('The number of dimensions to use when fitting the two '
+                       'matrices'),
+        'permutations': 'The number of permutations to be run when computing '
+                        'p-values. Supplying a value of zero will disable '
```
Comment (thermokarst) [inline q2_diversity/plugin_setup.py:272]:
```suggestion
                        'p-values. Supplying a value of `disable` will disable '
```

File: q2_diversity/tests/test_procrustes.py [hunk 4]
```diff
@@ -94,6 +109,45 @@ def test_procrustes(self):
...
+        skbio.util.assert_ordination_results_equal(other, self.expected_noise)
+
+        # the p value shouldn't be zero even in the presence of noise
+        self.assertAlmostEqual(true_m2, 0.7388121)
+        self.assertNotAlmostEqual(true_p_value, 0.001)
+
+    def test_zero_permutations_nan_pvalue(self):
+        ref, other, m2_results = procrustes_analysis(self.reference,
+                                                     self.other,
+                                                     permutations=0)
```
Comment (thermokarst) [inline q2_diversity/tests/test_procrustes.py:141]:
```suggestion
                                                     permutations='disable')
```

**Review by thermokarst — COMMENTED** [review]
This is great, @ElDeveloper! I think this should go in with this release (since it changes the format schema, better to get it in now, than later, and have to support multiple format versions). I proposed some changes inline, that get rid of the magic `0` value, and instead use the `'disable'` value instead. In general we've been trying to move away from magic numbers like `0` and `-1` for opting out of specific behaviors.

File: q2_diversity/plugin_setup.py (hunk 1 above)
Comment (ElDeveloper) [reply q2_diversity/plugin_setup.py:256]: THIS IS SO NEAT!

**Comment by ElDeveloper** [comment]
Thanks so much @thermokarst I didn't know about this.

**Comment by ElDeveloper** [comment]
Thanks @thermokarst, I was just in the process of fixing the broken test 👍

**Review by thermokarst — APPROVED** [review]

**Comment by thermokarst** [comment]
Thanks so much, @ElDeveloper! We'll need to update https://github.com/qiime2/q2-emperor/pull/89 with the new column name, I can do it, but if you get there before me, have at it.

**Pull request merged**